src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -602,6 +603,22 @@ public Function getGetterOrSetter(String name, int index, Scriptable scope, bool
         return isSetter ? slot.getSetterFunction(name, scope) : slot.getGetterFunction(name, scope);
     }
 
+    /**
+     * Get the getter or setter for a given property. Used by __lookupGetter__ and __lookupSetter__.
+     *
+     * @param name Name of the object. If nonnull, index must be 0.
+     * @param index Index of the object. If nonzero, name must be null.
+     * @param isSetter If true, return the setter, otherwise return the getter.
+     * @exception IllegalArgumentException if both name and index are nonnull and nonzero
+     *     respectively.
+     * @return Null if the property does not exist. Otherwise returns either the getter or the
+     *     setter for the property, depending on the value of isSetter (may be undefined if unset).
+     */
+    @Deprecated
```

Comment:
what about adding also a javadoc comment like this

* @deprecated since xxx please use .... instead

and maybe some works about the motivation of the new method and the drawbacks when still using this one (if there are any)